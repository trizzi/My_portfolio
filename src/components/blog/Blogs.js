import React from 'react';
import BlogItem from './BlogItem';
import blogLogo from '../../assets/about_button.svg';
import blogs from './blog.json';

const Blogs = () => {
	return (
		<div
			name='section5'
			data-testid='blog-section'
			className='bg-black px-4 py-10'>
			<div className='container-xl lg:container m-auto'>
				<div className='flex justify-center'>
					<img
						src={blogLogo}
						alt='blog-logo'
						className='h-24 rounded-full sm:shrink-0'
					/>
					<h1 className='text-4xl text-bold '>Blogs</h1>
				</div>

				<div className='grid grid-cols-1 md:grid-cols-3 gap-6'>
					{blogs.map((blog) => (
						<BlogItem
							key={blog.id}
							blog={blog}
						/>
					))}
				</div>
			</div>
		</div>
	);
};

export default Blogs;
