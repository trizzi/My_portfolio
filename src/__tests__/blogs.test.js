import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Blogs from '../components/blog/Blogs';

import { ThemeContextProvider } from '../context/ThemeContext';

// Mock the IntersectionObserver
beforeEach(() => {
	window.IntersectionObserver = jest.fn(function () {
		this.observe = jest.fn();
		this.disconnect = jest.fn();
		this.unobserve = jest.fn();
	});
});

// Mock the blog data
jest.mock('../components/blog/blog.json', () => [
	{ id: 1, title: 'Blog 1', content: 'Content 1', text: 'Text 1' },
	{ id: 2, title: 'Blog 2', content: 'Content 2', text: 'Text 2' },
	{ id: 3, title: 'Blog 3', content: 'Content 3', text: 'Text 3' },
]);

test('renders Blogs section with correct content', () => {
	const { container } = render(
		<ThemeContextProvider>
			<Blogs />
		</ThemeContextProvider>,
	);

	// Check if the Blogs section is rendered
	const blogsSection = screen.getByTestId('blog-section');
	expect(blogsSection).toBeInTheDocument();

	// Check if the blog items are rendered with the correct data
	const blogTitles = screen.getAllByRole('heading', { level: 3 }); // Assuming blog titles are rendered as h3 elements
	expect(blogTitles).toHaveLength(3); // Assuming there are three blog items

	// const blogContents = screen.getAllByText(/Content/i); // Matches any text containing "Content"
	// expect(blogContents).toHaveLength(3); // Assuming there are three blog items

	expect(container).toMatchSnapshot();
});
