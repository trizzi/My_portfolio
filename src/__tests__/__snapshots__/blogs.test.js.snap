// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Blogs section with correct content 1`] = `
<div>
  <div
    class="bg-black px-4 py-10"
    data-testid="blog-section"
    name="section5"
  >
    <div
      class="container-xl lg:container m-auto"
    >
      <div
        class="flex justify-center"
      >
        <img
          alt="blog-logo"
          class="h-24 rounded-full sm:shrink-0"
          src="about_button.svg"
        />
        <h1
          class="text-4xl text-bold "
        >
          Blogs
        </h1>
      </div>
      <div
        class="grid grid-cols-1 md:grid-cols-3 gap-6"
      >
        <div
          class="bg-white rounded-xl relative pb-6"
          style="opacity: 0; transform: translateY(100px) translateZ(0);"
        >
          <img
            alt="blog_image"
            class="h-[50%] w-full"
          />
          <div
            class="p-4"
          >
            <h3
              class="text-xl text-[#9932CC]"
            />
          </div>
          <div
            class="flex-grow border-t border-[#9932CC] shadow-xl "
          />
          <div
            class="text-black p-4"
          >
            Text 1...
          </div>
          <button
            class="text-violet-700 hover:text-violet-400 pl-4"
          >
            More
          </button>
          <div
            class="p-4 flex justify-end mb-32 md:mb-4"
          >
            <a>
              <button
                class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                style="background-color: transparent; color: rgb(0, 0, 0); border: 2px solid #9932cc; opacity: 0;"
                tabindex="0"
              >
                Read More
              </button>
            </a>
          </div>
        </div>
        <div
          class="bg-white rounded-xl relative pb-6"
          style="opacity: 0; transform: translateY(100px) translateZ(0);"
        >
          <img
            alt="blog_image"
            class="h-[50%] w-full"
          />
          <div
            class="p-4"
          >
            <h3
              class="text-xl text-[#9932CC]"
            />
          </div>
          <div
            class="flex-grow border-t border-[#9932CC] shadow-xl "
          />
          <div
            class="text-black p-4"
          >
            Text 2...
          </div>
          <button
            class="text-violet-700 hover:text-violet-400 pl-4"
          >
            More
          </button>
          <div
            class="p-4 flex justify-end mb-32 md:mb-4"
          >
            <a>
              <button
                class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                style="background-color: transparent; color: rgb(0, 0, 0); border: 2px solid #9932cc; opacity: 0;"
                tabindex="0"
              >
                Read More
              </button>
            </a>
          </div>
        </div>
        <div
          class="bg-white rounded-xl relative pb-6"
          style="opacity: 0; transform: translateY(100px) translateZ(0);"
        >
          <img
            alt="blog_image"
            class="h-[50%] w-full"
          />
          <div
            class="p-4"
          >
            <h3
              class="text-xl text-[#9932CC]"
            />
          </div>
          <div
            class="flex-grow border-t border-[#9932CC] shadow-xl "
          />
          <div
            class="text-black p-4"
          >
            Text 3...
          </div>
          <button
            class="text-violet-700 hover:text-violet-400 pl-4"
          >
            More
          </button>
          <div
            class="p-4 flex justify-end mb-32 md:mb-4"
          >
            <a>
              <button
                class="
			rounded-lg py-2 lg:py-2 px-5 whitespace-nowrap hover:bg-violet-400
        undefined undefined mt-5"
                style="background-color: transparent; color: rgb(0, 0, 0); border: 2px solid #9932cc; opacity: 0;"
                tabindex="0"
              >
                Read More
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
